--- a/thumbor/optimizers/gifv.py
+++ b/thumbor/optimizers/gifv.py
@@ -85,6 +85,7 @@
         if "webm" in self.context.request.filters:
             file_format = "webm"
             command_params = ["-quality", "good", "-cpu-used", "4"]
+            command_params += ["-c:v", "libvpx-vp9", "-b:v", "500K"]
         else:
             file_format = "mp4"
             command_params = ["-profile:v", "baseline", "-level", "4.0"]
