--- a/tests/loaders/test_http_loader.py
+++ b/tests/loaders/test_http_loader.py
@@ -366,7 +366,7 @@
         config = Config()
         config.HTTP_LOADER_CURL_ASYNC_HTTP_CLIENT = True
         config.HTTP_LOADER_CURL_LOW_SPEED_TIME = 1
-        config.HTTP_LOADER_CURL_LOW_SPEED_LIMIT = 1000000000000
+        config.HTTP_LOADER_CURL_LOW_SPEED_LIMIT = 1000000000
         ctx = Context(None, config, None)
 
         loader.load(ctx, url, self.stop)
@@ -389,7 +389,7 @@
         url = self.get_url('/')
         config = Config()
         config.HTTP_LOADER_CURL_LOW_SPEED_TIME = 1
-        config.HTTP_LOADER_CURL_LOW_SPEED_LIMIT = 1000000000000
+        config.HTTP_LOADER_CURL_LOW_SPEED_LIMIT = 1000000000
         ctx = Context(None, config, None)
 
         loader.load(ctx, url, self.stop)
