Description: Include webp files in MANIFEST.in
Author: Pablo Santiago Blum de Aguiar <scorphus@gmail.com>
Forwarded: not-needed
Last-Update: 2021-02-17
---
Index: app/MANIFEST.in
===================================================================
--- app.orig/MANIFEST.in
+++ app/MANIFEST.in
@@ -8,6 +8,7 @@ recursive-include thumbor *.jpg
 recursive-include thumbor *.png
 recursive-include thumbor *.gif
 recursive-include thumbor *.svg
+recursive-include thumbor *.webp
 recursive-include thumbor 513px-Coffee_beans_-_ziarna_kawy
 
 include LICENSE README.mkd CHANGELOG
